plugins/polls/app/views/polls/_form.html.haml

```diff
@@ -0,0 +1,76 @@
+- content_for :javascript do
+  :javascript
+    var choice = $($.parseHTML("#{escape_javascript(render(partial: 'choice', locals: {value: ''}))}"));
+
+    $(function() {
+      function updateSlideState() {
+        switch($('select[name="poll[voting_method]"]').val()) {
+        case 'event':
+        case 'single_select':
+          $('#multi_select_count').slideUp();
+          $('#min_max_points').slideUp();
+          break;
+
+        case 'multi_select':
+          $('#multi_select_count').slideDown();
+          $('#min_max_points').slideUp();
+          break;
+
+        case 'points':
+        case 'resistance_points':
+          $('#multi_select_count').slideUp();
+          $('#min_max_points').slideDown();
+          break;
+        }
+      }
+
+      $('select[name="poll[voting_method]"]').on('change', updateSlideState);
+
+      $('a[data-remove-choice]').on('click', function() {
+        $(this).parent().remove();
+        return false;
+      });
+
+      $('a[data-add-choice]').on('touchclick', function() {
+        choice.clone().appendTo('#choices');
+        return false;
+      });
+
+      updateSlideState();
+    });
```

Comment:
Let's put Javascript in an external file (unless it is really tiny or contains Ruby variables ).